internal/storage/image.go

```diff
@@ -398,6 +407,75 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, configImage, userSpecifiedImage string) error {
+	log.Debugf(ctx, "Getting image status of config image: %s", configImage)
+	configImageStatus, err := svc.ImageStatus(systemContext, configImage)
+	if err != nil {
+		return fmt.Errorf("image status from config image: %w", err)
+	}
+
+	log.Debugf(ctx, "Getting image status of user specified image: %s", configImage)
+	userSpecifiedImageStatus, err := svc.ImageStatus(systemContext, userSpecifiedImage)
+	if err != nil {
+		return fmt.Errorf("image status from user specified image: %w", err)
+	}
+
+	log.Debugf(ctx, "Comparing digest %s with %s, both must be equal.", configImageStatus.Digest, userSpecifiedImageStatus.Digest)
+	if configImageStatus.Digest != userSpecifiedImageStatus.Digest {
```

Comment:
- I don’t think this works reliably if `UserSpecifiedImage` is a moving tag. What really matters is whether the image we want to use was signed with the requested tag, more than whether the registry _currently_ points at the image we want to use, especially in the current model where Kubelet triggers a pull by name, gets an image ID, and then requests to run that ID, not the original name.
  - … and really I think this is unnecessary. Resolve the `configImage` into a specific `*storage.Image` or `*storage.Image.ID`, and then do both a signature check and container start using that image object. Resolving `userSpecifiedImage` into an image ID is not an essential part of the logic.
  - Alternatively, there is a good argument that for moving tags, what is on the registry matters, and that we should not blindly continue to use an older image if the on-registry tag has moved; but in that case this failure would need to trigger a re-pull to update the image!
- Comparing image `ID` is a bit more precise than `Digest`; several digests can be deduplicated to the same ID, and it’s not explicitly promised that `Image.Digest` will not change on such deduplication.

Reply:
I removed that check and did the `IsRunningImageAllowed` with `configImage`.

We would need the `userSpecifiedImage` only for moving tags, right?